catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/geofeature/GazetteerFeatureServiceTest.java

```diff
@@ -0,0 +1,110 @@
+import org.geotools.geometry.jts.spatialschema.geometry.primitive.PointImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.opengis.geometry.DirectPosition;
+
+public class GazetteerFeatureServiceTest {
+    private static final GeoEntry GEO_ENTRY_1 = new GeoEntry.Builder().name("Philadelphia")
+            .latitude(40)
+            .longitude(-71)
+            .featureCode("PPL")
+            .population(1000000)
+            .alternateNames("")
+            .build();
+
+    private static final GeoEntry GEO_ENTRY_2 = new GeoEntry.Builder().name("Camden")
+            .latitude(40)
+            .longitude(-70)
+            .featureCode("PPLC")
+            .population(10000000)
+            .alternateNames("")
+            .build();
+
+    private static final List<GeoEntry> QUERYABLE_RESULTS = Arrays.asList(GEO_ENTRY_1, GEO_ENTRY_2);
+
+    private static final String TEST_QUERY = "example";
+
+    private GazetteerFeatureService gazetteerFeatureService;
+
+    private GeoEntryQueryable geoEntryQueryable;
+
+    @Before
+    public void setUp() {
+        geoEntryQueryable = mock(GeoEntryQueryable.class);
+        gazetteerFeatureService = new GazetteerFeatureService();
+        gazetteerFeatureService.setGeoEntryQueryable(geoEntryQueryable);
+    }
+
+    @Test
+    public void testGetSuggestedFeatureNames() throws GeoEntryQueryException {
+        final int maxResults = 2;
+        doReturn(QUERYABLE_RESULTS).when(geoEntryQueryable)
+                .query(TEST_QUERY, maxResults);
+
+        List<String> results = gazetteerFeatureService.getSuggestedFeatureNames(TEST_QUERY,
+                maxResults);
+        assertThat(results.get(0), is(equalTo(GEO_ENTRY_1.getName())));
+        assertThat(results.get(1), is(equalTo(GEO_ENTRY_2.getName())));
+    }
+
+    @Test
+    public void testGetFeatureByName() throws GeoEntryQueryException {
+        final double north = 1;
+        final double south = -2;
+        final double east = 3;
+        final double west = -4;
+        GeoResult geoResult = new GeoResult();
+        geoResult.setFullName(GEO_ENTRY_1.getName());
+        final DirectPosition northWest = new DirectPositionImpl(west, north);
+        final DirectPosition southEast = new DirectPositionImpl(east, south);
+        geoResult.setBbox(Arrays.asList(new PointImpl(northWest), new PointImpl(southEast)));
+
+        gazetteerFeatureService = new GazetteerFeatureService() {
+            @Override
+            protected GeoResult getGeoResultFromGeoEntry(GeoEntry entry) {
+                return geoResult;
+            }
+        };
+        gazetteerFeatureService.setGeoEntryQueryable(geoEntryQueryable);
+
+        doReturn(QUERYABLE_RESULTS).when(geoEntryQueryable)
+                .query(TEST_QUERY, 1);
+
+        BoundingBoxFeature feature = (BoundingBoxFeature) gazetteerFeatureService.getFeatureByName(
+                TEST_QUERY);
+        assertThat(feature.getName(), is(equalTo(geoResult.getFullName())));
```

Comment:
nit : you could also just use is for all of these `assertThat(feature.getName(), is(geoResult.getFullName()));`